frontend_tests/node_tests/pm_list_data.js

```diff
@@ -0,0 +1,240 @@
+people.add_active_user(shiv);
+people.add_active_user(desdemona);
+people.add_active_user(lago);
+people.add_active_user(me);
+people.add_active_user(bot_test);
+people.add_active_user(jai);
+people.add_active_user(shivam);
+
+people.initialize_current_user(me.user_id);
+
+function get_list_info(zoomed) {
+    return pm_list_data.get_list_info(zoomed);
+}
+
+function test(label, f) {
+    run_test(label, ({override, override_rewire}) => {
+        pm_conversations.clear_for_testing();
+        pm_list.clear_for_testing();
+        f({override, override_rewire});
+    });
+}
+
+test("get_list_info_with_default_view", ({override}) => {
+    let list_info;
+    const timestamp = 0;
+
+    // we declare a empty list to make sure we don't have any PM thread present already.
+    const empty_list_info = get_list_info();
+
+    assert.deepEqual(empty_list_info, {
+        convos_to_be_shown: [],
+        more_convos_unread_count: 0,
+    });
+
+    // we would allot the number of unreads to always be 1 in each PM thread.
+    const num_unread_for_person = 1;
+
+    // initially we append 2 conversations and check for the `convos_to_be_shown` returned in list_info .
+    pm_conversations.recent.insert([101, 102], timestamp);
+    pm_conversations.recent.insert([103], timestamp);
+
+    override(unread, "num_unread_for_person", () => num_unread_for_person);
+
+    list_info = get_list_info();
+
+    const expected_list_info = [
+        {
+            is_active: false,
+            is_group: false,
+            is_zero: false,
+            recipients: "Aaron",
+            status_emoji_info: {
+                emoji_code: 20,
+            },
+            unread: 1,
+            url: "#narrow/pm-with/103-aaron",
+            user_circle_class: "user_circle_empty",
+            user_ids_string: "103",
+        },
+        {
+            recipients: "Alice, Bob",
+            user_ids_string: "101,102",
+            unread: 1,
+            is_active: false,
+            url: "#narrow/pm-with/101,102-group",
+            user_circle_class: undefined,
+            status_emoji_info: undefined,
+            is_group: true,
+            is_zero: false,
+        },
+    ];
+
+    assert.deepEqual(list_info, {
+        convos_to_be_shown: expected_list_info,
+        more_convos_unread_count: 0,
+    });
+
+    // Now we need to test that until the number of conversations reach `max_convos_to_show_with_unreads`
+    // we have all those conversations with unreads shown in default view itself.
+
+    // To check this we go with checking the number of unreads present in `more conversations` li-item.
+    pm_conversations.recent.insert([104], timestamp);
+    pm_conversations.recent.insert([103, 104], timestamp);
+    pm_conversations.recent.insert([105], timestamp);
+    pm_conversations.recent.insert([104, 105], timestamp);
+    pm_conversations.recent.insert([106], timestamp);
+    pm_conversations.recent.insert([105, 106], timestamp);
```

Comment:
Since sort order is a thing we want to be testing here, I think it'd be better to have different timestamps for different messages in this sort of test.

Reply:
Also, the second argument is a message ID, and `timestamp` is a bad variable name for that. I would consider just using the values 1-20 for message IDs here.

Finally, all the hardcoded 105s here should ideally be things like `jai.user_id` for readability.

Reply:
Hmm, updated the PR with this change!
thanks :)

Reply:
This does not appear updated at all.

Reply:
Sorry, i guess my terminal stopped the forced push due to fluctuation in network;

I think it should be visible now;